botorch/acquisition/multi_objective/joint_entropy_search.py

```diff
@@ -0,0 +1,739 @@
+
+        # Condition the model on the sampled pareto optimal points.
+        # TODO: Apparently, we need to make a call to the posterior otherwise
+        #  we run into a gpytorch runtime error:
+        #  "Fantasy observations can only be added after making predictions with a
+        #  model so that all test independent caches exist."
+
+        with fantasize_flag():
+            with settings.propagate_grads(False):
+                _ = self.initial_model.posterior(
+                    self.pareto_sets, observation_noise=False
+                )
+            # Condition with observation noise.
+            self.conditional_model = self.initial_model.condition_on_observations(
+                X=self.initial_model.transform_inputs(self.pareto_sets),
+                Y=self.pareto_fronts,
+            )
+
+    def _compute_posterior_statistics(
+        self, X: Tensor
+    ) -> dict[str, Union[Tensor, GPyTorchPosterior]]:
+        r"""Compute the posterior statistics.
+        Args:
+            X: A `batch_shape x q x d`-dim Tensor of inputs.
+
+        Returns:
+            A dictionary containing the posterior variables used to estimate the
+            entropy.
+
+            - "initial_entropy": A `batch_shape`-dim Tensor containing the entropy of
+                the Gaussian random variable `p(Y| X, D_n)`.
+            - "posterior_mean": A `batch_shape x num_pareto_samples x q x 1 x M`-dim
+                Tensor containing the posterior mean at the input `X`.
+            - "posterior_variance": A `batch_shape x num_pareto_samples x q x 1 x M`
+                -dim Tensor containing the posterior variance at the input `X`
+                excluding the observation noise.
+            - "observation_noise": A `batch_shape x num_pareto_samples x q x 1 x M`
+                -dim Tensor containing the observation noise at the input `X`.
+            - "posterior_with_noise": The posterior distribution at `X` which
+                includes the observation noise. This is used to compute the marginal
+                log-probabilities with respect to `p(y| x, D_n)` for `x` in `X`.
+        """
+        tkwargs = {"dtype": X.dtype, "device": X.device}
+        CLAMP_LB = torch.finfo(tkwargs["dtype"]).eps
+
+        posterior_statistics = dict()
+        # Compute the prior entropy term depending on `X`.
+        initial_posterior_plus_noise = self.initial_model.posterior(
+            X, observation_noise=True
+        )
+
+        # Additional constant term.
+        add_term = (
+            0.5
+            * self.model.num_outputs
+            * (1 + torch.log(2 * pi * torch.ones(1, **tkwargs)))
+        )
+        # The variance initially has shape `batch_shape x (q*M) x (q*M)`
+        # prior_entropy has shape `batch_shape`.
+        initial_entropy = add_term + 0.5 * torch.logdet(
+            initial_posterior_plus_noise.mvn.covariance_matrix
+        )
+
+        posterior_statistics["initial_entropy"] = initial_entropy
```

Comment:
```suggestion
        # Compute the prior entropy term depending on `X`.
        initial_posterior_plus_noise = self.initial_model.posterior(
            X, observation_noise=True
        )

        # Additional constant term.
        add_term = (
            0.5
            * self.model.num_outputs
            * (1 + torch.log(2 * pi * torch.ones(1, **tkwargs)))
        )
        # The variance initially has shape `batch_shape x (q*M) x (q*M)`
        # prior_entropy has shape `batch_shape`.
        initial_entropy = add_term + 0.5 * torch.logdet(
            initial_posterior_plus_noise.mvn.covariance_matrix
        )

        posterior_statistics = {"initial_entropy": initial_entropy}
```
A Meta-internal linter complains about the `dict()` call, and also let's define `posterior_statistics` closer to where it's used